### PR TITLE
system: more runnableExamples + doc improvements

### PR DESCRIPTION
**Pull request by timotheecour**

### CODE REVIEW

File: lib/system.nim [hunk 1]
```diff
@@ -729,37 +728,47 @@ proc len*(x: cstring): int {.magic: "LengthStr", noSideEffect.} =
...
+    type Foo = enum
+      f0 = 0, f1 = 3
+    assert f1.ord == 3
+    type Bar = distinct int
+    assert 3.Bar.ord == 3
+
+func chr*(u: range[0..255]): char {.magic: "Chr".} =
+  ## Converts `u` to a `char`, same as `char(u)`.
+  runnableExamples:
+    doAssert chr(65) == 'A'
```
Comment (dom96) [inline lib/system.nim:764]: I wonder if we should be consistent everywhere and use `doAssert` instead of `assert` above?

**Review by dom96 — APPROVED** [review]
Looks good.

File: lib/system.nim (hunk 1 above)
Comment (timotheecour) [reply lib/system.nim:764]: system.nim uses a lot of both doAssert and assert in runnableExamples, but I agree there should be 1 preferred way; we should settle which one is preferred via a `nep1` PR (unlike for tests where doAssert is clearly better, there are arguments on both sides for runnableExamples); I'm leaning towards `assert`.

Comment (timotheecour) [reply lib/system.nim:764]: => followup https://github.com/nim-lang/Nim/pull/17077

**Pull request merged**